File: AngularVsJS/Models/Position.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using AngularVsJS.Models.Enums;

namespace AngularVsJS.Models
{
    public class Position
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public PositionType Type { get; set; }

        [Required]
        public PositionStatus Status { get; set; }

        public string Description { get; set; }

        [Required]
        [DataType(DataType.Date)]
        public DateTime Date { get; set; }
    }
}

File: AngularVsJS/ClientApp/app/models/position.ts
import {PositionType} from "./enums/positionType";
import { PositionStatus } from "./enums/positionStatus";

export interface IPosition {
    id: string;
    type: PositionType;
    name: string;
    date: Date;
    description?: string;
    status: PositionStatus;
}

export class PositionItem implements IPosition {
    public id: string;
    public type: PositionType;
    public name: string;
    public date: Date;
    public description?: string;
    public status: PositionStatus;
}


